AirCasting/ABConnector/SDSyncController.swift

```diff
@@ -0,0 +1,28 @@
+// Created by Lunar on 17/11/2021.
+//
+
+import Foundation
+import CoreBluetooth
+
+class SDSyncController: ObservableObject {
+    private let fileWriter: SDSyncFileWriter
+    private let airbeamServices: SDCardAirBeamServices
+//    private let airbeamConnection: CBPeripheral
+    
+    init(airbeamServices: SDCardAirBeamServices, fileWriter: SDSyncFileWriter) {
+        self.airbeamServices = airbeamServices
+        self.fileWriter = fileWriter
+    }
+    
+    func syncFromAirbeam(_ airbeamConnection: CBPeripheral, completion: @escaping (Bool) -> Void) {
+        airbeamServices.downloadData(from: airbeamConnection, progress: { [weak self] chunk in
+            // Filesystem write
+            self?.fileWriter.writeToFile(data: chunk.payload, sessionType: chunk.sessionType)
```

Comment:
note that this writes are synchronous. I think we should schedule them to a different thread so that it won't block the bluetooth connection